keanu-project/src/test/java/io/improbable/keanu/network/BayesianNetworkTest.java

```diff
@@ -152,6 +152,24 @@ public void networkReturnsVerticesInNamespace() {
         assertTrue(verticesInNamespace.containsAll(Arrays.asList(a0, a1, b0)));
     }
 
+    @Test
+    public void networkReturnsAllWithLabelInnerLabel() {
+        BooleanVertex a0 = new BernoulliVertex(0.5);
+        BooleanVertex a1 = new BernoulliVertex(0.5);
+        BooleanVertex b0 = new BernoulliVertex(0.5);
+        BooleanVertex c = new BernoulliVertex(0.5);
+
+        a0.setLabel(new VertexLabel("root", "a", "0"));
+        a1.setLabel(new VertexLabel("root", "a", "1"));
+        b0.setLabel(new VertexLabel("root", "b", "0"));
```

Comment:
`setLabel` returns `this` so you can do it in one line, e.g.:

    BooleanVertex a0 = new BernoulliVertex(0.5).setLabel(new VertexLabel("root", "a", "0"));